File: frontend/src/types/duplicates.ts
import type { MqttMessage } from './mqtt';

export interface DuplicatePair {
  topics: [string, string];
  scores: [number, number];
}

export interface DuplicateData {
  data1: MqttMessage[];
  data2: MqttMessage[];
}

export type ConfirmDuplicateRequest =
  | { action: 'keep'; topics: [string, string] }
  | { action: 'delete'; topics: [string] };

File: backend/models/duplicates.py
from pydantic import BaseModel
from typing import List, Tuple, Literal

class GraphPoint(BaseModel):
    timestamp: int
    value: float

class DuplicatesResponse(BaseModel):
    data1: List[GraphPoint]
    data2: List[GraphPoint]

class DuplicatePair(BaseModel):
    topics: Tuple[str, str]
    score: float

class ConfirmDuplicateRequest(BaseModel):
    action: Literal['keep', 'delete']
    topics: List[str]